tests/scripts/thread-cert/command.py

```diff
@@ -324,11 +323,22 @@ def check_child_id_response(command_msg, route64 = CheckType.OPTIONAL, network_d
     check_mle_optional_tlv(command_msg, active_operational_dataset, mle.ActiveOperationalDataset)
     check_mle_optional_tlv(command_msg, pending_operational_dataset, mle.PendingOperationalDataset)
 
-    # Check the detail of network_data only when the CheckType is CONTAIN
-    # and the check function is set
-    if network_data == CheckType.CONTAIN and network_data_detail_check_func != None:
-        network_data_tlv = command_msg.assertMleMessageContainsTlv(mle.NetworkData)
-        network_data_detail_check_func(network_data_tlv)
+def check_child_id_response_stable_data(msg):
+    """Verify a properly formatted Child Id Response to request of stable network data
+    """
+    check_child_id_response(msg)
+    network_data_tlv = msg.assertMleMessageContainsTlv(mle.NetworkData)
+    prefixes = [tlv for tlv in network_data_tlv.tlvs if isinstance(tlv, network_data.Prefix)]
+    assert len(prefixes) == 1, 'Network data should only contain one prefix tlv here.'
+    check_prefix_and_border_router_16(prefixes[0], 0xFFFE)
+
+def check_child_id_response_full_data(msg):
+    check_child_id_response(msg)
+    network_data_tlv = msg.assertMleMessageContainsTlv(mle.NetworkData)
+    prefixes = [tlv for tlv in network_data_tlv.tlvs if isinstance(tlv, network_data.Prefix)]
+    assert len(prefixes) >= 2, 'Network data should have at least two prefixes here.'
```

Comment:
the number `2` is too specific for this case, better to design a more generic method as previous comment